tests/benchmark/test_analog_stream_readers.py

```diff
@@ -7,81 +7,95 @@
 from nitypes.waveform import AnalogWaveform
 from pytest_benchmark.fixture import BenchmarkFixture
 
-import nidaqmx
+from nidaqmx import Task
+from nidaqmx.constants import WaveformAttributeMode
 from nidaqmx.stream_readers._analog_multi_channel_reader import AnalogMultiChannelReader
 from nidaqmx.stream_readers._analog_single_channel_reader import (
     AnalogSingleChannelReader,
 )
 
 
 @pytest.mark.benchmark(group="analog_stream_readers")
-def test___analog_single_channel_reader___read_one_sample___1_sample(
-    benchmark: BenchmarkFixture,
-    ai_single_channel_task: nidaqmx.Task,
+@pytest.mark.parametrize("num_channels", [1])
+@pytest.mark.parametrize("num_samples", [1])
+def test___analog_single_channel_reader___read_one_sample(
+    benchmark: BenchmarkFixture, ai_benchmark_task: Task, num_channels: int, num_samples: int
 ) -> None:
-    reader = AnalogSingleChannelReader(ai_single_channel_task.in_stream)
+    reader = AnalogSingleChannelReader(ai_benchmark_task.in_stream)
 
     benchmark(reader.read_one_sample)
 
 
 @pytest.mark.benchmark(group="analog_stream_readers")
-def test___analog_single_channel_reader___read_many_sample___1000_samples(
-    benchmark: BenchmarkFixture,
-    ai_single_channel_task: nidaqmx.Task,
+@pytest.mark.parametrize("num_channels", [1])
+@pytest.mark.parametrize("num_samples", [2, 1000])
```

Comment:
Use [1, 1000] so you can compare read_one_sample against read_many_sample with num_samples=1.

I assume read_many_sample doesn't have some weird restriction where it doesn't support 1. The sample clock timing properties have that restriction but read should not.